Sources/Protocols/MessagesLayoutDelegate.swift

```diff
@@ -81,6 +81,25 @@ public protocol MessagesLayoutDelegate: AnyObject {
     ///   The default value returned by this method is zero.
     func messageBottomLabelHeight(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> CGFloat
 
+    /// Specifies the size for the `MessageContentCell`'s accessory view.
+    ///
+    /// - Parameters:
+    ///   - message: The `MessageType` that will be displayed for this cell.
+    ///   - messagesCollectionView: The `MessagesCollectionView` in which this cell will be displayed.
+    ///
+    /// - Note:
+    ///   The default value returned by this method is zero.
+    func accessoryViewSize(for message: MessageType, in messagesCollectionView: MessagesCollectionView) -> CGSize
```

Comment:
I'm not sure if we want to make this a delegate method or to put it as a property of the layout object.
@zhongwuzw, this is a feature MessageKit should support so I'd like your advice?

Reply:
I guess the delegate calls should be different by indexPath's section or item? I could make the change

Reply:
Ah but by moving it to size calculator, we'd need another display delegate method like the `func configureAvatarView(_ avatarView: AvatarView, for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView)`, and then do something like `if let layout = messagesCollectionView.collectionViewLayout as? MessagesCollectionViewFlowLayout {
            layout.textMessageSizeCalculator.incomingAvatarSize = CGSize(width: 52, height: 52)
            layout.textMessageSizeCalculator.outgoingAvatarSize = CGSize(width: 52, height: 52)
            layout.textMessageSizeCalculator.incomingAvatarPosition = AvatarPosition(vertical: .messageCenter)
            layout.textMessageSizeCalculator.outgoingAvatarPosition = AvatarPosition(vertical: .messageCenter)
        }`, which I personally feel pretty weird. Let me know what you think

Reply:
The reason for this is I really wanted to move away from having users configure things with the delegates. I'd rather they abstract the logic for the layout of a cell into a `CellSizeCalculator` object. Then provide the layout object with the calculator.

IMO, doing it another way does not feel right. The API we're using here feels like it was designed for an earlier version of MessageKit and not version 1.0

Reply:
@SD10 , My some thoughts, wether we can provide a flag to not use `accessoryView`, not just set size to `.zero`, because I think many of the time, maybe user don't need `accessoryView` at all.

Reply:
@zhongwuzw what if we just make the default size for it `.zero`?

Reply:
@zhongwuzw this is true 👍 but I've learned not to prematurely optimize. I only optimize if someone complains and then if we do benchmarks afterward. We do cell layout using frames and not auto-layout so we already save so much. This is likely the last view we will support. I will not add views that don't exist in JSQMVC